projects/sandbox/datagen/datagen/utils/timeslides.py

```diff
@@ -31,13 +31,26 @@ def __init__(
         self.signal_times = np.arange(
             start + buffer, stop - buffer - max_shift, spacing
         )
+        self.buffer = buffer
+        self.max_shift = max_shift
         self.jitter = jitter
 
     @property
     def num_signals(self):
         return len(self.signal_times)
 
     def __call__(self, waveform_duration: float):
+        min_buffer = waveform_duration + self.jitter - self.max_shift
```

Comment:
Because if `min_buffer = waveform_duration + jitter - max_shift`, then `stop - min_buffer - max_shift = stop - (waveform_duration + jitter - max_shift) - max_shift = stop - waveform_duration - jitter`, which is the least amount of time we need to guarantee that our signals fit.

Reply:
Ok but let's say `max_shift > 0`, then `min_buffer < waveform_duration + jitter`, which means we could in principle sample a jitter that puts some of the waveform before the signal starts?

Reply:
Could you elaborate on what you mean by putting some of the waveform before the signal starts? I'm not seeing what you're saying

Reply:
I guess let's take the concrete example of the values we use for vanilla BBHNet:
```
waveform_duration = 8
jitter = 1
max_shift = 10
```
Then in this case, `min_buffer = -1`, so a buffer of `0` would be acceptable. But then in the worst case if we sample `jitter = -1` for the first signal, the trigger time gets placed at `start + waveform_duration / 2 - 1`, which means the first second of the signal falls before `start`.

Would it make more sense to just pass `waveform_duration` to `__init__` (since this never changes for us anyway) and doing all the checks there? Or rather even than doing checks, we could do something like
```python
# build the length of the waveform and the jitter into
# these properties to ensure that we'll never be too
# close to the edge or one another. Redefine buffer to
# be the minimum required distance from any part of
# the signal to the edge of the timeseries, and spacing
# to be the minimum required space between the end
# of one signal and the last, and have these default to 0
buffer = waveform_duration // 2 + jitter + buffer
spacing = waveform_duration + 2 * jitter + spacing
self.signal_times = np.arange(start + buffer, stop - max_shift - buffer, spacing)
```
Then just have a `__call__` method that accepts no args, and where we'll always know that no value of `jitter` could violate our spacing constraints (also realized that we can save a tiny bit of space by doing `jitter = np.random.uniform(-1, 1, self.num_signals) * self.jitter`)

Reply:
Hmm, yeah, I like this better. It's neater, and the variables are more interpretable too. I'll get that implemented.